lib/audited/auditor.rb

```diff
@@ -220,30 +222,21 @@ def write_audit(attrs)
       end
 
       def presence_of_audit_comment
-        case
-        when !auditing_enabled
-          return true
-        when audit_comment.present?
-          return true
-        when audited_options[:on].exclude?(:create) && self.new_record?
-          return true
-        when audited_options[:on].exclude?(:update) && self.persisted?
-          return true
-        when audited_changes.empty? && self.persisted?
-          return true
-        else
-          errors.add(:audit_comment, "can't be blank.")
-        end
+        require_comment if comment_required_state?
       end
 
       def require_comment
         if auditing_enabled && audit_comment.blank?
-          errors.add(:audit_comment, "Comment required before destruction")
-          return false if Rails.version.start_with?('4.')
-          throw :abort
+          errors.add(:audit_comment, "Comment can't be blank!")
+          throw(:abort)
         end
       end
 
+      def comment_required_state?
+        (audited_options[:on].include?(:create) && self.new_record?) ||
```

Comment:
Right. So, here we are with what I was talking about the validity of the object while in a given state, but not being created / updated. My reasoning is that we only care about the existence of the comment when:

1.) The object is a new record and we require a comment + audit on create

2.) The object is an existing record, has had some attributes changed, and we require a comment + audit on update.